### ISSUE TITLE
Give progress information on long running commands

### ISSUE THREAD

**Issue by robert-ancell** [canonical/ubuntu-pro-client#1814]
Some commands like `ua enable` can take a long time to complete. It would be useful to express this on the command line so a graphical application can inform the user about this.

Possible solution:
```
$ sudo ua enable foo --format=json
{"progress": 10, "status-code": "checking", "status": "Checking for foo service"}
{"progress": 30, "status-code": "enabling", "status": "Enabling foo"}
{"progress": 60, "status-code": "enabling", "status": "Enabling foo"}
{"progress": 90, "status-code": "enabling", "status": "Enabling foo"}
{"progress": 95, "status-code": "finalizing", "status": "Finalizing"}
{"progress": 100, "status-code": "done", "status": "Done"}
```

**Comment by orndorffgrant** [comment]
Tracking in SC-430

**Comment by orndorffgrant** [comment]
I mis-read this issue as another one when I changed the labels earlier.

This is implemented for the API versions of attach and enable using the `--show-progress` flag for their API functions.
For example, `sudo pro api --show-progress u.pro.services.enable.v1 --args service=esm-infra` will give json lines of each step.

However, I totally failed to document this :facepalm: I opened a new issue to represent the documentation we need to write https://github.com/canonical/ubuntu-pro-client/issues/3315
